api/account_nonce/module.go

```diff
@@ -46,8 +47,17 @@ func (m Module) AccountNonce(dataPtr int32, dataLen int32) int64 {
 	b := m.memUtils.GetWasmMemorySlice(dataPtr, dataLen)
 	buffer := bytes.NewBuffer(b)
 
-	publicKey := types.DecodePublicKey(buffer)
-	nonce := m.systemModule.Get(publicKey).Nonce
+	publicKeyDec, err := types.DecodePublicKey(buffer)
+	if err != nil {
+		log.Critical(err.Error())
+		return 0
```

Comment:
this `return` will never be executed since `log.Critical` first logs and then calls `panic` at the end